jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/ResourceContentFactory.java

```diff
@@ -91,10 +91,14 @@ private HttpContent load(String pathInContext, Resource resource)
             {
                 String compressedPathInContext = pathInContext + format.getExtension();
                 Resource compressedResource = this._factory.newResource(compressedPathInContext);
-                if (compressedResource != null && compressedResource.exists() && compressedResource.lastModified().isAfter(resource.lastModified()) &&
-                    compressedResource.length() < resource.length())
-                    compressedContents.put(format,
-                        new ResourceHttpContent(compressedResource, _mimeTypes.getMimeByExtension(compressedPathInContext)));
+                if (compressedResource == null || !compressedResource.exists())
+                    continue; // skip, doesn't exist
+                if (compressedResource.lastModified().isBefore(resource.lastModified()))
+                    continue; // skip, compressed resource modified before actual resource
+                if (compressedResource.length() >= resource.length())
+                    continue; // skip, not smaller than actual resource
+                // add found precompressed format
```

Comment:
Remove the comments, they are just describing what the code does, and the code is very clear.
Less chances that the comments rot.